### ISSUE TITLE
Notification sound effect not playing in certain locales (eg: Russian)

### ISSUE THREAD

**Issue by Zren** [Zren/plasma-applet-eventcalendar#209]
Using my `translate/plasmoidlocaletest ru` script, I was able to reproduce the bug that multiple users are experiencing.

* driglu4it on Element
* lipun on https://store.kde.org/p/998901/
* #162 may be related, though it could've been reported before the notification code was refactored.

```
qml: cmd [python3,/home/chris/Code/plasma-applets/eventcalendar/package/contents//scripts/notification.py,--app-name,Таймер,--icon,chronometer,--sound,/usr/share/sounds/freedesktop/stereo/complete.oga,--action,repeat,Повтор,--metadata,1610145336724,Таймер - Стоп,0:01 прошло]
qml:    args.summary Таймер - Стоп => Таймер - Стоп
qml:    args.body 0:01 прошло => 0:01 прошло
qml:    cmd.done
qml:       cmd 'python3' '/home/chris/Code/plasma-applets/eventcalendar/package/contents//scripts/notification.py' '--app-name' 'Таймер' '--icon' 'chronometer' '--sound' '/usr/share/sounds/freedesktop/stereo/complete.oga' '--action' 'repeat,Повтор' '--metadata' '1610145336724' 'Таймер - Стоп' '0:01 прошло'
qml:       exitCode 0
qml:       exitStatus 0
qml:       stdout usage: notification.py [-h] [--icon ICON] [--app-name APPNAME] [--sound SOUND]
                       [--loop LOOP] [--action ACTIONS] [--metadata METADATA]
                       summary message

Notifications with sound effects and actions.

positional arguments:
  summary
  message

optional arguments:
  -h, --help           show this help message and exit
  --icon ICON
  --app-name APPNAME
  --sound SOUND
  --loop LOOP
  --action ACTIONS
  --metadata METADATA

qml:       stderr 'latin-1' codec can't encode characters in position 0-5: ordinal not in range(256)
```

The `qml:       stderr 'latin-1' codec can't encode characters in position 0-5: ordinal not in range(256)` error is likely from the `libcanberra.so` code.

**Comment by Zren** [comment]
Hmm, changing 

```python
def convertArgs(args):
	return (
		arg.encode("latin-1") if isinstance(arg, str) else arg
		for arg in args
	)
```

to `arg.encode("utf-8")` fixes the bug. Not sure why I used `latin-1` there so I need to make sure before pushing the change.

**Comment by Zren** [comment]
Running `watch -n0.3 'pactl list sink-inputs | grep "application.name"'` then activeing the Timer complete displayed `application.name = "Таймер"` which is the Russian translation for `msgid "Timer"` so it appears to accept the `utf-8` string.

**Comment by madjoe** [comment]
Hi, currently I use v73, but I'm still facing this issue. The sound is set to the default `/usr/share/sounds/freedesktop/stereo/complete.oga`, but the play button is disabled in settings and there's no sound after the timer reaches 0.

**Comment by Zren** [comment]
Did the timer sound effect work in the past?

Are you using a russian locale (or another non ASCII locale)?

Have you restarted Plasma after updating?

Do these commands work?

```
cd ~/.local/share/plasma/plasmoids/org.kde.plasma.eventcalendar/contents/scripts/
python3 ./notification.py summary message --sound complete
python3 ./notification.py summary message --sound /usr/share/sounds/freedesktop/stereo/complete.oga
python3 ./notification.py summary message --sound file:///usr/share/sounds/freedesktop/stereo/complete.oga
```

**Comment by Zren** [comment]
> hr_HR.UTF-8 UTF-8

This works for me, so I don't think it's the locale.

```
LANGUAGE="hr_HR:hr" LANG="hr_HR.UTF-8" LC_TIME="hr_HR.UTF-8" python3 ./notification.py summary message --sound file:///usr/share/sounds/freedesktop/stereo/complete.oga
```

> No. Each of those `python3` lines kept hanging without producing any sound. I had to `Ctrl+C` to exit shortly after executing each of those commands.

The commands will wait until you close the notification (prints `close` to stdout), or click it (`default`) before exiting. Are you sure it's hanging because of the sound effect?

-----

I assume you have `libcanberra` installed since it's a dependency of the volume widget (`plasma-pa`).

```
$ pacman -Ss libcanberra
extra/libcanberra 0.30+2+gc0620e4-3 [installed]    A small and lightweight implementation of the XDG Sound Theme Specification
```

Are you certain that `libcanberra.so.0` exists?

```bash
$ locate libcanberra.so.0
# ... steam
# ... lutris
/usr/lib/libcanberra.so.0
/usr/lib/libcanberra.so.0.2.5
/usr/lib32/libcanberra.so.0
/usr/lib32/libcanberra.so.0.2.5
```

Are you certain that `/usr/share/sounds/freedesktop/stereo/complete.oga` exists?

```bash
$ locate complete.oga
# ... steam
/usr/share/sounds/freedesktop/stereo/complete.oga
```

**Comment by madjoe** [comment]
> The commands will wait until you close the notification (prints `close` to stdout), or click it (`default`) before exiting. Are you sure it's hanging because of the sound effect?

Sorry, my bad - I've just noticed I had "Do not disturb" enabled for months and I completely became unaware it's not the default. Now, when I enabled notifications, I can see the popup when timer reaches 0 as well as I can hear the notification sound. Sorry for raising a false alarm! :(

**Comment by Zren** [comment]
lmao :laughing:, well at least you found out other notifications weren't working either. Cheers.

**Comment by madjoe** [comment]
> lmao , well at least you found out other notifications weren't working either. Cheers.

For some reason, I'm under impression that even with "Do not disturb" enabled, the Timer sound previously worked just fine, which was the exact setup I wanted. :)

**Comment by Goldfive** [comment]
> Did the timer sound effect work in the past?
> 
> Are you using a russian locale (or another non ASCII locale)?
> 
> Have you restarted Plasma after updating?
> 
> Do these commands work?
> 
> ```
> cd ~/.local/share/plasma/plasmoids/org.kde.plasma.eventcalendar/contents/scripts/
> python3 ./notification.py summary message --sound complete
> python3 ./notification.py summary message --sound /usr/share/sounds/freedesktop/stereo/complete.oga
> python3 ./notification.py summary message --sound file:///usr/share/sounds/freedesktop/stereo/complete.oga
> ```

These all work fine with me. I don't have Do not Disturb enabled.. and I also do not get a notification. I'm using Plasma 5.22.2 with the new notification panel. It worked somewhere in the past... but I'm only now spending time on getting it to work again. The timer displaying a notification and sound would be great - but even the calendar itself it not pushing events, not even for future reference.

**Comment by Zren** [comment]
@Goldfive: Anything show up in the logs? `journalctl -b0 -f`

**Comment by Goldfive** [comment]
> @Goldfive: Anything show up in the logs? `journalctl -b0 -f`

Well, not really anything specific to this issue. Also the play button when choosing sounds for the timer is disabled... no clue if this has anything to do with this. It might be I have a specific setting set somewhere else... but it's quite strange since other widgets are able to send notifications & sounds. I did a grep on calendar and this showed up: https://pastebin.com/LPj32YG7

**Comment by Goldfive** [comment]
I found it. Apparently `python3` isn't available in the context path of the widget. Because when I ran it using `plasmawindowed`, `plasmavoidviewer` or just replacing the `plasmashell` using the terminal, it always worked (thanks to your widgets guide btw :)). 

Now I use zsh where I put all of my paths in.. and it could be that the `/usr/bin` path just isn't accessible at that point in time in bash. This definitely is a mistake of mine - and I managed to fix it. This also made me very curious about plasmoid widgets.. so this is great :dancing_men: Thanks for the swift response :+1:
